duecredit/tests/test_io.py

```diff
@@ -192,6 +192,72 @@ def test_output():
     assert_equal(modules['package.module'][0],
                  collector.citations[('package.module', entry.get_key())])
 
+
+def test_output_return_all():
+    entry = BibTeX(_sample_bibtex)
+    entry2 = BibTeX(_sample_bibtex2)
+
+    # normal use
+    collector = DueCreditCollector()
+    collector.cite(entry, path='package')
+    collector.cite(entry2, path='package2')
+
+    output = Output(None, collector)
+
+    packages, modules, objects = output._get_collated_citations(tags=['*'])
+    assert_false(packages)
+    assert_false(modules)
+    assert_false(objects)
+
+    for flag in ['1', 'True', 'TRUE', 'true', 'on', 'yes']:
+        with patch.dict(os.environ, {'DUECREDIT_REPORT_ALL': flag}):
+            # if _all is None then get the environment
+            packages, modules, objects = output._get_collated_citations(tags=['*'])
+            assert_equal(len(packages), 2)
```

Comment:
if we are talking about 'pleasing', here I would have checked for exact values of packages, not just that there is 2 of them.  
also ideally we should have extracted the function for getting a bool value out of env variable for `_ALL` and tested it properly testing for 'true' value while not bothering here with a full sweep across all possible values here. but I guess since test is cheap, it is ok'ish